x/ecocredit/server/testsuite/suite.go

```diff
@@ -103,12 +104,29 @@ func (s *IntegrationTestSuite) TestScenario() {
 	// create batch
 	t0, t1, t2 := "10.37", "1007.3869", "0"
 	tSupply0 := "1017.7569"
-	r0, r1, r2:= "4.286", "10000.4589902", "0"
+	r0, r1, r2 := "4.286", "10000.4589902", "0"
 	rSupply0 := "10004.7449902"
 
+	time1 := time.Now()
+	time2 := time.Now()
+
+	// Batch creation should fail if the EndDate is before the StartDate
```

Comment:
Let's also add some tests cases for where one or both dates are omitted and verify that it fails. We could even just do these against Msg.ValidateBasic